zebedee-cms/src/test/java/com/github/onsdigital/zebedee/api/PageTest.java

```diff
@@ -350,10 +387,11 @@ public void testPage_deletePage_collectionNotFound() throws ZebedeeException, IO
     public void testPage_deletePage_pageHookException() throws ZebedeeException, IOException {
 
         // Given a mock page hook that throws an exception
-        PageUpdateHook mockPageHook = mock(PageUpdateHook.class);
-        doThrow(new RuntimeException()).when(mockPageHook).onPageUpdated(any(), anyString());
+        PageUpdateHook<com.github.onsdigital.zebedee.content.page.base.Page> mockPageHook = mock(PageUpdateHook.class);
```

Comment:
Unfortunately we have 2 different `Page` objects, and one being the class in test so I had to use the fully qualified name for the other one

Reply:
As discussed:
This is a valid question and normally I would advise avoiding naming 2 classes the same if they are both within our control and rename if there is a sensible alternative. However, renaming in this case would be a pretty painful experience and it would **have** to be the Page object not the Page API class that would need to change.

For context: Restolino API endpoints are registers by applying the `@Api` annotation to a class. Restolino uses the class name as the uri path the endpoint will be registered this is something you cannot change/reconfigure. For example:

> Annotating `Page.java` with `@Api` will create an api endpoint that is registered for requests to uri `/Page`.  If you renamed this to `PageEndpoint.java` the endpoint would be registered for requests to `/PageEndpoint`. This would have an impact on Babbage and the scope of the change would continue to ripple out. 

I will never understand or agree with this Restolino decision, IMO its stupid, inflexible and completely impractical.

For this case the path of least resistance is use the fully qualified path name to differentiate between the two different `Path` objects as Rafa has done. This isn't pretty but refactoring this would be a sizeable change with a high risk of regression - we don't current have sufficient test coverage/quality to give us confidence the change was working as expected.